server/src/main/java/org/apache/druid/server/coordinator/rules/Rule.java

```diff
@@ -52,5 +51,14 @@
 
   boolean appliesTo(Interval interval, DateTime referenceTimestamp);
 
+  /**
+   * {@link DruidCoordinatorRuntimeParams#getAvailableSegments()} must not be called in Rule's code, because the
+   * available segments are not specified for the {@link DruidCoordinatorRuntimeParams} passed into Rule's code. This is
+   * because {@link DruidCoordinatorRuntimeParams} entangles two slightly different (nonexistent yet) abstractions:
+   * "DruidCoordinatorHelperParams" and "RuleParams" which contain params that only {@link
```

Comment:
Could you introduce `RuleParams` class which doesn't contain `getAvailableSegments` method?

Reply:
Maybe I'll do that in a big refactoring that aims to fix #7159 and #7202.